ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -34,13 +39,15 @@
 class Log implements Write {
   private final String dataDir;
   private final long capacityInBytes;
+  private final boolean isLogSegmented;
   private final StoreMetrics metrics;
+  private final Iterator<Pair<String, String>> segmentNameAndFileNameIterator;
```

Comment:
what is the purpose of this iterator ? I don't see it getting updated in first constructor? Am I reading it wrongly